mobile/src/main/java/rs/readahead/washington/mobile/views/activity/CollectFormEntryActivity.java

```diff
@@ -326,8 +326,10 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
 
     @NeedsPermission(Manifest.permission.ACCESS_FINE_LOCATION)
     public void startPermissionProcess() {
-        manageLocationSettings(C.GPS_PROVIDER, () -> {
+        checkLocationSettings(C.GPS_PROVIDER, () -> {
```

Comment:
is this handling the timeout issue or not?

Reply:
Yes, if needed the showGpsMetadataDialog is called which has the maybeChangeTemporaryTimeout method.